tests/unit/cmd/scan/test_path.py

```diff
@@ -368,6 +368,26 @@ def test_scan_path_should_detect_non_git_files(self, cli_fs_runner):
         ), "not_committed files not should have been ignored"
         assert result.exception is None
 
+    @my_vcr.use_cassette()
+    def test_scan_path_use_gitignore(self, cli_fs_runner: CliRunner) -> None:
+        """
+        GIVEN a directory containing a gitignore
+        WHEN executing a scan with --use-gitignore option
+        THEN ggshield should not scan the ignored files
+        """
+        os.makedirs("repo_git_ignore")
+        Path("repo_git_ignore/secret.js").write_text(VALID_SECRET_PATCH)
```

Comment:
It would be a better test if the repository also contained one secret to be found. This way the test proves that `--use-gitignore` does not cause all files to be ignored 😉.